src/SpectatorView.Unity/Assets/SpectatorView/Scripts/StateSynchronization/StateSynchronizationBroadcaster.cs

```diff
@@ -181,8 +180,7 @@ public void OnFrameCompleted()
                     message.Write(StateSynchronizationObserver.PerfCommand);
                     StateSynchronizationPerformanceMonitor.Instance.WriteMessage(message, numFrames);
                     message.Flush();
-                    var data = memoryStream.ToArray();
-                    connectionManager.Broadcast(ref data);
+                    connectionManager.Broadcast(memoryStream.GetBuffer(), 0, memoryStream.Position);

```

Comment:
Adding a helper method that accepts a memory stream could be a better idea than repeating the same pattern multiple times (especially if Flush is actually required, see the other comments).

Reply:
I'll hold off on this for now. I've opened the following issue to look into this: https://github.com/microsoft/MixedReality-SpectatorView/issues/307

Reply:
I don't think #307 is related. It's not about reusing the streams or writers, it's about making sure that the same error-prone pattern is not repeated multiple times. I don't think it's a blocker if all the usages are fixed here, though.